### PR TITLE
Chores: remove fileshare client from cloud provider

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Chores: remove fileshare client in cloud provider
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by MartinForReal** [comment]
releated: https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/2117

**Review by andyzhangx — APPROVED** [review]
/lgtm

File: pkg/provider/azure_file.go [hunk 1]
```diff
@@ -1,43 +0,0 @@
...
-	"sigs.k8s.io/cloud-provider-azure/pkg/azureclients/fileclient"
-)
-
-// create file share
-func (az *Cloud) createFileShare(ctx context.Context, subsID, resourceGroupName, accountName string, shareOptions *fileclient.ShareOptions) error {
-	_, err := az.FileClient.WithSubscriptionID(subsID).CreateFileShare(ctx, resourceGroupName, accountName, shareOptions, "")
-	return err
-}
-
-func (az *Cloud) deleteFileShare(ctx context.Context, subsID, resourceGroupName, accountName, name string) error {
```
Comment (andyzhangx) [inline pkg/provider/azure_file.go:33]: we need subsID parameter in azure file interfaces, could you add that parameter back? @MartinForReal 
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/82e716a56f865e021ce9860c6c7d5dae294409d0/pkg/azureclients/fileclient/azure_fileclient.go#L132

Comment (andyzhangx) [reply pkg/provider/azure_file.go:33]:
```
func (az *Cloud) createFileShare(ctx context.Context, subsID, resourceGroupName, accountName string, shareOptions *fileclient.ShareOptions) error {
	_, err := az.FileClient.WithSubscriptionID(subsID).CreateFileShare(ctx, resourceGroupName, accountName, shareOptions, "")
	return err
}

func (az *Cloud) deleteFileShare(ctx context.Context, subsID, resourceGroupName, accountName, name string) error {
	return az.FileClient.WithSubscriptionID(subsID).DeleteFileShare(ctx, resourceGroupName, accountName, name, "")
}

func (az *Cloud) resizeFileShare(ctx context.Context, subsID, resourceGroupName, accountName, name string, sizeGiB int) error {
	return az.FileClient.WithSubscriptionID(subsID).ResizeFileShare(ctx, resourceGroupName, accountName, name, sizeGiB)
}

func (az *Cloud) getFileShare(ctx context.Context, subsID, resourceGroupName, accountName, name string) (storage.FileShare, error) {
	return az.FileClient.WithSubscriptionID(subsID).GetFileShare(ctx, resourceGroupName, accountName, name, "")
}
```

**Pull request merged**